### PR TITLE
Add pytest for descriptions

### PR DESCRIPTION
**Pull request by christophfroehlich**
I added some simple tests to this repo. 

I copied them over from https://github.com/UniversalRobots/Universal_Robots_ROS2_Description

~~Locally, `colcon test` does not run it, why?~~ I had to manually install `pip install -U pytest`, otherwise an old version got installed causing import errors.

### CODE REVIEW

**Review by destogl — DISMISSED** [review]

**Comment by destogl** [comment]
Can we actually rather add integration tests? This file might be an interesting inspiration:

https://github.com/StoglRobotics-forks/kuka_experimental/pull/12/files#diff-b2da8ae937165518fe9c1d9f7fc39094e9c0fd179eb1389115ca444936f06a3c

File: example_1/test/test_view_robot_launch.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+@pytest.mark.rostest
+def generate_test_description():
+    launch_include = IncludeLaunchDescription(
+        PythonLaunchDescriptionSource(
+            os.path.join(
+                get_package_share_directory("ros2_control_demo_example_1"),
+                "launch/view_robot.launch.py",
+            )
+        ),
+        launch_arguments={"gui": "true"}.items(),
```
[the remark this answers is not in the document]
Comment (christophfroehlich) [reply example_1/test/test_view_robot_launch.py:51]: no this is also tested with the CI workflow, see the result here: https://github.com/ros-controls/ros2_control_demos/actions/runs/6094788897/job/16537030308?pr=342#step:6:1621

**Comment by christophfroehlich** [comment]
> Can we actually rather add integration tests? This file might be an interesting inspiration:
> 
> https://github.com/StoglRobotics-forks/kuka_experimental/pull/12/files#diff-b2da8ae937165518fe9c1d9f7fc39094e9c0fd179eb1389115ca444936f06a3c

This PR is the first step, just testing if xacro is happy with the urdf and if the view_robot.launch.py is starting without any errors. 
I'll have a look, thanks for the link.

**Comment by christophfroehlich** [comment]
@mergifyio backport iron humble

**Review by saikishor — APPROVED** [review]
Good work. LTGM
Thank you

**Review by saikishor — APPROVED** [review]
LGTM! Thanks for the recent changes

File: example_9/test/test_view_robot_launch.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
+# Copyright (c) 2022 FZI Forschungszentrum Informatik
```
Comment (olivier-stasse) [inline example_9/test/test_view_robot_launch.py:1]: Dear @christophfroehlich thanks a lot for this very nice addition.
However I believe that all the licenses part is a copy pase that should be removed.
First it is a BSD Clause 2 license, while APACHE is the license claimed in the root license file.
Second the author is somebody else...

As this repository is for people learning ros2_control and that unit test is very important, removing this unnecessary part will help to keep the example clear.

Again thanks a lot for this welcome addition.

File: example_6/test/test_view_robot_launch.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
+# Copyright (c) 2022 FZI Forschungszentrum Informatik
```
Comment (olivier-stasse) [inline example_6/test/test_view_robot_launch.py:1]: Please remove license,  just add instition and author name, and refer to the license in the repository's root.

File: example_7/test/test_urdf_xacro.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
+# Copyright (c) 2022 FZI Forschungszentrum Informatik
```
Comment (olivier-stasse) [inline example_7/test/test_urdf_xacro.py:1]: Please remove license,  just add instition and author name, and refer to the license in the repository's root.

File: example_7/test/test_view_robot_launch.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
+# Copyright (c) 2022 FZI Forschungszentrum Informatik
```
Comment (olivier-stasse) [inline example_7/test/test_view_robot_launch.py:1]: Please remove license,  just add instition and author name, and refer to the license in the repository's root.

File: example_8/test/test_urdf_xacro.py [hunk 6]
```diff
@@ -0,0 +1,77 @@
+# Copyright (c) 2022 FZI Forschungszentrum Informatik
```
Comment (olivier-stasse) [inline example_8/test/test_urdf_xacro.py:1]: Please remove license,  just add instition and author name, and refer to the license in the repository's root.

File: example_8/test/test_view_robot_launch.py [hunk 7]
```diff
@@ -0,0 +1,54 @@
+# Copyright (c) 2022 FZI Forschungszentrum Informatik
```
Comment (olivier-stasse) [inline example_8/test/test_view_robot_launch.py:1]: Please remove license,  just add instition and author name, and refer to the license in the repository's root.

**Review by olivier-stasse — COMMENTED** [review]
Dear @christophfroehlich thanks for the PR.
I believe that the license is not correct.

File: example_9/test/test_view_robot_launch.py (hunk 2 above)
Comment (christophfroehlich) [reply example_9/test/test_view_robot_launch.py:1]: Thanks for the hint, I'm not sure about the license stuff and how ament_copyright deals with it.. But I can try ;) You mean that I just remove the license part but let the line with the copyright claim?

**Pull request merged**